0012-common-ledger-api/0012-common-ledger-api.md

```diff
@@ -0,0 +1,1181 @@
+The message body should be a JSON object with one field, `fulfillment`, which contains a [Crypto-Condition Fulfillment][] in string format.
+
+#### Response Format
+
+A successful result uses the HTTP response code **201 Created**. The message body of the response is a JSON object containing the submitted fulfillment in the `fulfillment` field. A ledger MUST return a successful response if and only if the transfer was executed as a result of this request.
+
+If you
+
+#### Example
+
+Request:
+
+```json
+PUT /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204/fulfillment
+Content-Type: application/json
+
+{
+  "fulfillment": "cf:0:_v8"
+}
+```
+
+Response:
+
+```json
+HTTP/1.1 200 OK
+
+{
+  "fulfillment": "cf:0:_v8"
+}
+```
+
+#### Errors
+
+- [NotFoundError][] - The transfer does not exist.
+- [UnmetConditionError][] - The fulfillment does not match the condition.
+- [TransferNotConditionalError][] - The transfer had no condition to fulfill.
+- [TransferStateError][] - The transfer was not in the `prepared` state when the request was received. This occurs if the transfer has already been executed, rejected, or expired.
+- [UnprocessableEntityError][] - The request is formatted properly but contains an otherwise-unspecified semantic problem.
+- [InvalidBodyError][] - The request body was not properly-formatted JSON.
+
+
+### Reject Transfer
+[Reject Transfer]: #reject-transfer
+
+Reject a prepared transfer. A transfer can be rejected if and only if that transfer is in the `prepared` state. Doing so transitions the transfer to the `rejected` state.
+
+**Authorization:** The `credit_account` of a transfer MUST be able to reject the transfer. Unauthorized users and the `debit_account` MUST NOT be able to reject the transfer. Other account owners MUST NOT be able to reject the transfer unless those accounts are explicitly authorized to do so by the `credit_account` and those accounts are not the `debit_account`.
+
+#### Request Format
+
+PUT to the `transfer_rejection` path of the transfer you want to execute, with a JSON object in the message body containing a rejection reason.
+
+**Metadata URL Name:** `transfer_rejection`. Suggested generic path: `/transfers/{client_id}/rejection`
+
+##### Body Parameters
+
+The message body should be a JSON object with one top-level field:
+
+| Field              | Value  | Description                                    |
+|:-------------------|:-------|:-----------------------------------------------|
+| `rejection_reason` | String | The reason for rejecting the transfer. This is intended to be a machine-readable identifier. |
+
+#### Response Format
+
+A successful result uses the HTTP response code **200 OK** and contains a JSON object containing the updated [Transfer resource][] containing the rejection reason.
+
+#### Example
+
+Request:
+
+```json
+PUT /transfers/3a2a1d9e-8640-4d2d-b06c-84f2cd613204/rejection
+Content-Type: application/json
+
+{
+  "rejection_reason": "BlacklistedSender"
+}
+```
+
+Response:
```

Comment:
I'd be willing to change it. The rationale behind the suggested behavior is that the rejection reason is generally represented as one field of the transfer object. The request here is the only case where the rejection reason appears outside of the transfer object. We can keep the number of returned resources low if we return the whole transfer object. This also has the added benefit of providing the full context and updated state of the transfer right away, which makes one usage pattern (which I expect to be common) more efficient -- the case where you get the updated transfer state immediately after rejecting a transfer. I think that's likely to happen a lot because apps may want to show an updated date/status thing for the transfer. If it can save the response to "Reject Transfer" then that saves us a full round trip.

Any further opinions one way or the other on this?